### PR TITLE
fix-default-byte-as-binary-payload-in-modular

### LINKED ISSUE #2082: Content-type of application/octet-stream is being base64 encoded...
We have TypeSpec that looks like:
```
op analyzeFromStream is Azure.Core.RpcOperation<
  {
    @doc("The format of the HTTP payload.")
    @header
    contentType: "application/octet-stream";

    @doc("The image to be analyzed")
    @body
    imageContent: bytes;
...
```

And it's producing TypeScript:
```
    .post({
      ...operationOptionsToRequestParameters(options),
      contentType: (options.contentType as any) ?? "application/octet-stream",
      queryParameters: { mode: mode },
      body: uint8ArrayToString(imageContent, "base64"),
    });
```

### PR DESCRIPTION
**Pull request by qiaozha**
fixes #2082

### CODE REVIEW

File: packages/typespec-ts/test/modularUnit/apiOperations.spec.ts [hunk 1]
```diff
@@ -0,0 +1,224 @@
+import { assert } from "chai";
+import { emitModularOperationsFromTypeSpec } from "../util/emitUtil.js";
+import { assertEqualContent } from "../util/testUtil.js";
+
+describe.only("api operations in Modular", () => {
```
Comment (MaryGao) [inline packages/typespec-ts/test/modularUnit/apiOperations.spec.ts:5]: skip only

File: packages/typespec-ts/src/modular/helpers/operationHelpers.ts [hunk 2]
```diff
@@ -407,8 +408,9 @@ function buildBodyParameter(
           bodyParameter.clientName
         }, "${getEncodingFormat(bodyParameter.type)}")`;
   } else if (
-    bodyParameter.type.type === "byte-array" &&
-    bodyParameter.type.format === "binary"
+    (bodyParameter.type.type === "byte-array" &&
+      bodyParameter.type.format === "binary") ||
+    bodyParameter.isBinaryPayload
```
Comment (MaryGao) [inline packages/typespec-ts/src/modular/helpers/operationHelpers.ts:413]: It seems that when `bodyParameter.type.type === "byte-array" &&
      bodyParameter.type.format === "binary"` is also binary, why not enhance the isBinaryPayload logic to cover that case?

**Review by MaryGao — APPROVED** [review]
Approved and left minor comments

File: packages/typespec-ts/test/modularUnit/apiOperations.spec.ts [hunk 3]
```diff
@@ -0,0 +1,224 @@
+import { assert } from "chai";
+import { emitModularOperationsFromTypeSpec } from "../util/emitUtil.js";
+import { assertEqualContent } from "../util/testUtil.js";
+
+describe("api operations in Modular", () => {
+  it("should handle contentTypes has binary data", async () => {
+    const tspContent = `
+    @route("/uploadFileViaBody")
+    @post op uploadFileViaBody(
+      @header contentType: "application/octet-stream",
```
Comment (lirenhe) [inline packages/typespec-ts/test/modularUnit/apiOperations.spec.ts:10]: From Swagger, the payload of octet-stream would look like the below, so could you help to ensure the typespec is correct for this case?

requestBody:
  content:
    application/octet-stream:
      schema:
        type: string
        format: binary # binary data
    image/png:
      schema:
        type: string
        format: base64 # base64-encoded data

**Review by lirenhe — APPROVED** [review]

File: packages/typespec-ts/src/utils/modelUtils.ts [hunk 4]
```diff
@@ -83,7 +83,10 @@ export function isByteOrByteUnion(dpgContext: SdkContext, type: Type) {
 }
 
 function isByteType(schema: any) {
-  return schema.type === "string" && schema.format === "byte";
+  return (
+    schema.type === "string" &&
+    (schema.format === "byte" || schema.format === "binary")
```
Comment (qiaozha) [inline packages/typespec-ts/src/utils/modelUtils.ts:88]: If it's self defined scalar, we will get the format of binary as special handled.

File: packages/typespec-ts/src/modular/helpers/operationHelpers.ts (hunk 2 above)
Comment (qiaozha) [reply packages/typespec-ts/src/modular/helpers/operationHelpers.ts]: Changed both to use isBinaryPayload and fixed a small bug in isByteType.

File: packages/typespec-ts/test/modularUnit/apiOperations.spec.ts (hunk 3 above)
Comment (qiaozha) [reply packages/typespec-ts/test/modularUnit/apiOperations.spec.ts]: Issue created https://github.com/microsoft/typespec/issues/2615

**Pull request merged**